### ISSUE TITLE
RFC: Standard model for gap junctions

### ISSUE THREAD

**Issue by apdavison** [NeuralEnsemble/PyNN#434]
There is now sufficient simulator support to justify adding official support for gap junctions in PyNN.

Here is a proposal for the API. Please comment below.

1) gap junctions should be defined as for other types of synapses, i.e. by passing an instance of a subclass of `StandardSynapseType` to a `Projection`.
2) possible names for this subclass are `ElectricalSynapse`, `GapJunction`, `GapJunctionSynapse`.
3) the standard model will have one parameter, `weight`, which is the bidirectional conductance in µS.
4) Creating a gap junction connection should install _both_ halves of the gap junction. In the case of NEST, for example, this means that each PyNN connection will comprise two NEST connections (a-->b and b-->a)

Example usage:

```python
gap_junction = sim.GapJunction(weight=0.1)
connections = sim.Projection(population1, population2,
                             sim.AllToAllConnector(),
                             synapse_type=gap_junction)
```

**Comment by MRIO** [comment]
Hi Andrew!

Are you also considering gap junctions that implement particular functions for conductances? For instance, i use (matlab anonymous function):

`fgap = @(DeltaV) (0.8 .* exp(-1.*DeltaV.*DeltaV/100) + 0.2);`

Which has been used in Schweighofer et al., 1999 and other places.

**Comment by tmchartrand** [comment]
Interested to see any updates to this. Am I correct that there's currently some unsupported functionality for gap junctions with NEURON only?

**Comment by apdavison** [comment]
@tmchartrand yes, there is a class `pyNN.neuron.ElectricalSynapse`, contributed by @tclose about four years ago, but I'm not sure if it still works. The reason for creating this RFC was that NEST has now added a model for gap junctions, and Brian also supports writing gap junction models.

I do not expect to have time to work on this in the near future, but I'd be happy to support anyone who would like to work on it (the first step would be to investigate whether the NEST model is the same as that in `pyNN.neuron.ElectricalSynapse`).

**Comment by apdavison** [comment]
I've just realized that I have not replied to the question from @MRIO :

It would certainly be possible to support more complex gap junction models such as you describe. The general rule in PyNN is that a "standard model" must be implemented by at least two simulators.
